pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -419,8 +420,14 @@ func (p *provider) getConfig(s v1alpha1.ProviderSpec) (*Config, *providerconfigt
 		}
 		iops := *rawConfig.DiskIops
 
-		if iops < 100 || iops > 64000 {
-			return nil, nil, nil, errors.New("Invalid value for `diskIops` (min: 100, max: 64000)")
+		if c.DiskType == ec2.VolumeTypeIo1 {
+			if iops < 100 || iops > 64000 {
+				return nil, nil, nil, errors.New("Invalid value for `diskIops` (min: 100, max: 64000)")
+			}
+		} else if c.DiskType == ec2.VolumeTypeGp3 {
+			if iops < 3000 || iops > 64000 {
+				return nil, nil, nil, errors.New("Invalid value for `diskIops` (min: 3000, max: 64000)")
+			}
 		}
```

Comment:
This needs to be moved out of the if block. Right now it looks like:

```
if c.DiskType == ec2.VolumeTypeIo1 {
	//....
	if c.DiskType == ec2.VolumeTypeIo1 {
	//....
	} else if c.DiskType == ec2.VolumeTypeGp3 {
	//....
	}
	//....
}
```

You'll never hit the condition `else if c.DiskType == ec2.VolumeTypeGp3`

Reply:
Aha! good catch, thanks. I actually did not see the outer if condition.

Reply:
Makes you wonder how many issues you can introduce with 10 lines of change ...

Reply:
Should be fixed now! PTAL.